src/containers/blocks.jsx

```diff
@@ -328,7 +328,10 @@ class Blocks extends React.Component {
             error.message = `Workspace Update Error: ${error.message}`;
             log.error(error);
         }
-        this.workspace.addChangeListener(this.props.vm.blockListener);
+        // All of the changes that happened during the load above are queued with
+        // timeouts, so re-enable the listener in the next tick, so it happens after
+        // the events are already fired.
+        setTimeout(() => this.workspace.addChangeListener(this.props.vm.blockListener));
```

Comment:
Does this work? I thought we tried and it didn't...

Reply:
Something about this plus pulling out the extraneous targetUpdates in the vm when you switch sprites made it start working for me.